fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -1175,6 +1185,54 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             !processIntent(intent)
     }
 
+    private fun showFullscreenMessageSurveyIfNeeded(context: Context) {
+        val messagingStorage = context.components.analytics.messagingStorage
+        val messages = runBlockingIncrement { messagingStorage.getMessages() }
+        val nextMessage =
+            messagingStorage.getNextMessage(FenixMessageSurfaceId.SURVEY, messages)
+                ?: return
+
+        if (nextMessage.id == "fullscreen-viewpoint-invitation-message") {
+            val currentBootUniqueIdentifier = BootUtils.getBootIdentifier(context)
```

Comment:
I see what you mean.  Since we're not expected to show it again.  I'll still set it just in case something changed and we didn't notice that it's not set.  That said, I'll move it back since this is not needed until we mark the message as displayed.